pages/cookbook/index.mdx

```diff
@@ -114,30 +114,216 @@ dump("The loop is over!");
+    }
+
+    // Internal message receiver, which responds to a String message "pop"
+    receive("pop") {
+        // Remove the last item and decrease the counter
+        self.arrPop();
+    }
+
+    // Internal message receiver, which responds to a String message "get_last"
+    receive("get_last") {
+        // Reply back with the latest item in the map if it exists
+        if (self.arrLength > 0) {
+            // Note, that we use !! operator as we know for sure that the value would be there
+            let last: Int = self.arr.get(self.arrLength - 1)!!;
+            self.reply(last.toCoinsString().asComment());
+        }
+    }
+
+    // Getter function for obtaining the «array»
+    get fun map(): map<Int, Int> {
+        return self.arr;
+    }
+
+    // Getter function for obtaining the current length of the «array»
+    get fun length(): Int {
+        return self.arrLength;
+    }
+}
+```
+
+<Callout type="warning" emoji="⚠️">
+
+  Note, that manually keeping track of number of items or checking the length of such map is very error-prone and generally discouraged. Instead, try to wrap your map into the [Struct](/book/structs-and-messages#structs) and define [extension functions](/book/functions#extension-function) on it. See the [next example showcasing such approach](#how-to-emulate-an-array-using-a-map-wrapped-in-a-struct).
+
+</Callout>
+
+### How to emulate an array using a map wrapped in a Struct
+
+```tact
+import "@stdlib/deploy"; // for Deployable trait
+
+struct Array {
+    map: map<Int, Int>; // «array» of Int values as a map of Ints to Ints
+    length: Int = 0;    // length of the «array», defaults to 0
+}
+
+// Extension mutation function for adding new entries
+extends mutates fun push(self: Array, item: Int) {
+    self.map.set(self.length, item); // set the entry (key-value pair)
+    self.length += 1;                // increase the length field
+}
+
+// Extension mutation function for deleting the last entry and returning its value
+extends mutates fun pop(self: Array): Int {
+    require(self.length > 0, "No items in the array to delete!");
+
+    // Note, that we use !! operator as we know for sure that the value would be there
+    let lastItem: Int = self.map.get(self.length - 1)!!;
+    self.map.set(self.length - 1, null); // delete the entry
+    self.length -= 1;                    // decrease the length field
+
+    return lastItem;
+}
+
+// Extension function for returning the last value
+extends fun getLast(self: Array): Int {
+    require(self.length > 0, "No items in the array!");
+
+    // Note, that we use !! operator as we know for sure that the value would be there
+    return self.map.get(self.length - 1)!!;
+}
+
+contract MapsAsArrays with Deployable {
+    // Persistent state variables
+    array: Array; // our «array», which uses the map
+
+    // Constructor (initialization) function of the contract
+    init() {
+        self.array = Array{map: emptyMap()}; // length defaults to 0
```

Comment:
We should define an `emptyArray` constructor: it should look like a map, as if arrays are also builtin.

Reply:
Ok, will do!